library/src/main/scala/scala/scalajs/js/ArrayOps.scala

```diff
@@ -98,7 +83,7 @@ object ArrayOps {
     throw new UnsupportedOperationException(msg)
 
   /** Non-inlined implementation of [[ArrayOps.++]]. */
-  private def concat[A](left: Array[_ <: A], right: Array[_ <: A]): Array[A] = {
+  private def scalajsConcat[A](left: Array[_ <: A], right: Array[_ <: A]): Array[A] = {
```

Comment:
`concat` is now defined in the new collections. We have to use a different name.